src/AutoRest.CSharp/Common/Input/InputTypes/InputType.cs

```diff
@@ -1,9 +1,11 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
+using System.Security.Cryptography.X509Certificates;
+
 namespace AutoRest.CSharp.Common.Input;
 
-internal abstract record InputType(string Name, bool IsNullable)
+internal abstract record InputType(string Name, bool IsNullable, string? OriginalName = default)
```

Comment:
so we added a `OriginalName` for all the types to store the name before we do any transformations.
It does not look elegant - but I believe we need it because we have to write the swagger names such as when `mgmt-debug.show-serialized-names` is on.
If there are other approaches, we might want to try that, such as let MPG output library to hold a mapping of name transformations?
It is ok to keep as is

Reply:
If we can have a rename map somewhere, we don't even need to make settable name properties in InputType